src/strategies/ejs-staking/index.ts

```diff
@@ -0,0 +1,250 @@
+// Subgraph query
+const getGraphQLQuery = (addresses, blockTag) => {
+
+  // Ignore block if latest block
+  const block = blockTag === 'latest' ? null : { number: blockTag };
+
+  return {
+    endusers: {
+      __args: {
+        where: {
+          id_in: addresses
+        },
+        block
+      },
+      id: true,
+      stakes: {
+        stakingPool: {
+          id: true
+        },
+        stakeAmountWei: true,
+        stakeMaturityTimestamp: true,
+        isSuspended: true
+      }
+    }
+  };
+};
+
+// Path to identify unique settings pool
+const SEPERATOR = '|';
+const encodePath = (networkId, poolId) => [networkId, poolId].join(SEPERATOR);
+
+// Types
+interface Params {
+  stakingServiceContractAddress: string,
+  voteEndDate: string,
+  decimals, number,
+  stakingPoolSettings: Array<StakingPoolSetting>,
+  legacyData: Array<StakeInfo>
+};
+
+interface StakingPoolSetting {
+  pools: Array<Pool>
+  lotSizePerVote: string,
+  minimumTokensToVote: string,
+};
+
+interface Pool {
+  poolId: string,
+  networkId: string,
+  legacy: boolean
+};
+
+interface StakeInfo {
+  address: string,
+  networkId: string,
+  poolId: string,
+  stakeAmountWei: string,
+  stakeMaturityTimestamp: string,
+  isSuspended: boolean
+};
+
+interface ProcessedStakeInfo {
+  address: string,
+  settingsIndex: number,
+  stakeAmountWei: BigNumber,
+  stakeMaturityTimestamp: BigNumber,
+  isSuspended: boolean
+};
+
+// Strategy
+export async function strategy(
+  space,
+  network,
+  provider,
+  addresses,
+  options: Params,
+  snapshot
+): Promise<Record<string, number>> {
+
+  // Map networkId and poolId to each setting
+  const { stakingPoolSettings } = options;
+
+  const settingsIndices: Record<string, number> = Object.fromEntries(
+    stakingPoolSettings.flatMap((stakingPoolSetting, index) => {
+      return stakingPoolSetting.pools.map(pool => {
+        const path = encodePath(pool.networkId, pool.poolId);
+        return [path, index];
+      });
+    })
+  );
+
+  // Get list of networkIds
+  const networkIds: Array<string> = stakingPoolSettings
+    .flatMap(setting => setting.pools)
+    .map(pool => pool.networkId)
+    // unique
+    .filter((element, index, array) => {
+      return array.indexOf(element) == index;
+    });
+
+  // Get blocktag for for each network id based on blocktag of main network
+  const mainBlockTag = typeof snapshot === 'number' ? snapshot : 'latest';
+
+  const blocks = await getSnapshots(
+    network,
+    mainBlockTag,
+    provider,
+    networkIds
+  );
+
+  const blockArray = Object.entries(blocks);
+
+  // Call graph for each network
+  const promises = blockArray.map(([network, block]) => {
+
+    // Get url of subgraph
+    const url = EJS_SUBGRAPH_URL[network];
+
+    // If subgraph doesn't exist, we skip
+    if (url === undefined) return null;
+
+    // Call graph with query
+    const query = getGraphQLQuery(addresses, block);
+    return subgraphRequest(EJS_SUBGRAPH_URL[network], query);
+  });
```

Comment:
Hi @ChaituVR you can pass any number of arrays into the pools subArray but the number of requests is capped by the number of networks configured in the EJS_SUBGRAPH_URL constant